File: app-front-survey/src/service/ISurveyService.ts
import { SurveyCreate } from "../dto/request/survey/survey.create";
import { Survey } from "../dto/response/survey/survey";

export interface ISurveyService {
    list: () => Promise<Survey[]>;
    create: (data: SurveyCreate) => Promise<Survey>;
    findById: (surveyId: number) => Promise<Survey>;
}

File: app-back-survey/src/main/java/com/eduardo/survey/service/ISurveyService.java
package com.eduardo.survey.service;

import com.eduardo.survey.dto.request.survey.ReqSurveyCreate;
import com.eduardo.survey.entity.Survey;

import java.util.List;

public interface ISurveyService {
    Survey create(ReqSurveyCreate surveyDto);

    List<Survey> list();

    Survey findById(Long surveyId);

    Short countIpSurvey(Long surveyId, String ip);

}
